socorro/unittest/processor/test_processor_app.py

```diff
@@ -185,97 +185,120 @@ def mocked_save_raw_and_processed(*_):
         config.logger.info.assert_called_with(
             'Error captured in Sentry! Reference: someidentifier'
         )
-        assert len(captured_exceptions) == 2
-        captured_exception, captured_exception_2 = captured_exceptions
-        assert captured_exception.__class__ == NameError
-        assert captured_exception.args[0] == 'waldo'
-        assert captured_exception_2.__class__ == AssertionError
-        assert captured_exception_2.args[0] is False
 
     @mock.patch('socorro.lib.raven_client.raven')
-    def test_transform_misc_error_with_raven_configured_successful(
-        self,
-        mock_raven,
-    ):
-
-        captured_exceptions = []  # a global
-
+    def test_transform_save_error_with_raven_configured_successful(self, mock_raven):
+        # Mock everything
         def mock_capture_exception(exc_info=None):
-            captured_exceptions.append(exc_info)
             return 'someidentifier'
 
         raven_mock_client = mock.MagicMock()
         raven_mock_client.captureException.side_effect = mock_capture_exception
-
         mock_raven.Client.return_value = raven_mock_client
 
-        config = self.get_standard_config(
-            sentry_dsn='https://abc123@example.com/project'
-        )
+        # Set up a processor and mock .save_raw_and_processed() to raise an exception
+        config = self.get_standard_config(sentry_dsn='https://abc123@example.com/project')
         pa = ProcessorApp(config)
         pa._setup_source_and_destination()
         pa.source.get_raw_crash.return_value = DotDict({'raw': 'crash'})
         pa.source.get_raw_dumps_as_files.return_value = {}
 
+        expected_exception = ValueError('simulated error')
+
         def mocked_save_raw_and_processed(*_):
-            raise ValueError('Someone is wrong on the Internet')
+            raise expected_exception
 
-        pa.destination.save_raw_and_processed.side_effect = (
-            mocked_save_raw_and_processed
-        )
-        # The important thing is that this is the exception that
-        # is raised and not something from the raven error handling.
+        pa.destination.save_raw_and_processed.side_effect = mocked_save_raw_and_processed
+
+        # Run .transform() and make sure it raises the ValueError
         with pytest.raises(ValueError):
             pa.transform('mycrashid')
 
+        # Assert that we sent the exception to Sentry
+        raven_mock_client.captureException.assert_called_once_with(expected_exception)
+
+        # Assert that the logger logged the appropriate thing
+        config.logger.info.assert_called_with(
+            'Error captured in Sentry! Reference: someidentifier'
+        )
+
+    @mock.patch('socorro.lib.raven_client.raven')
+    def test_transform_get_error_with_raven_configured_successful(self, mock_raven):
+        # Mock everything
+        def mock_capture_exception(exc_info=None):
+            return 'someidentifier'
```

Comment:
Since this is now just returning a string, we can just use the `return_value` attribute on the mock and avoid defining an extra function at all.